### DISCUSSION TITLE
-S100 but result is 200k

### DISCUSSION THREAD

**Discussion by guettli** [tjko/jpegoptim#100] [General]
```
jpegoptim --stdout -S100k orig.jpg > result-jpegoptim.jpg 
```

```
===> ls -lh result-jpegoptim.jpg
-rw-r--r-- 1 guettli guettli 201K Mai 17 09:31 result-jpegoptim.jpg
```

```
===> jpegoptim --version
jpegoptim v1.4.4  x86_64-pc-linux-gnu
Copyright (c) 1996-2016, Timo Kokkonen

libjpeg version: 8d  15-Jan-2012
Copyright (C) 1991-2017 The libjpeg-turbo Project and many others
```

![orig](https://user-images.githubusercontent.com/414336/82138628-2da01f80-9822-11ea-99b9-f7ce24004f56.jpg)

![result-jpegoptim](https://user-images.githubusercontent.com/414336/82138608-f9c4fa00-9821-11ea-95f8-66510cc69879.jpg)

**Comment by zvezdochiot** [comment]
```sh
$ jpegquality result-jpegoptim.jpg

jpegquality version 1.2

#File: result-jpegoptim.jpg
Quantization table
  Precision=0; Table index=0 (luminance)
     255 255 255 255 255 255 255 255
     255 255 255 255 255 255 255 255
     255 255 255 255 255 255 255 255
     255 255 255 255 255 255 255 255
     255 255 255 255 255 255 255 255
     255 255 255 255 255 255 255 255
     255 255 255 255 255 255 255 255
     255 255 255 255 255 255 255 255
  Estimated quality level = -155.00%
  Quality too low; estimate may be incorrect.
Quantization table
  Precision=0; Table index=1 (chrominance)
     255 255 255 255 255 255 255 255
     255 255 255 255 255 255 255 255
     255 255 255 255 255 255 255 255
     255 255 255 255 255 255 255 255
     255 255 255 255 255 255 255 255
     255 255 255 255 255 255 255 255
     255 255 255 255 255 255 255 255
     255 255 255 255 255 255 255 255
  Estimated quality level = -155.00%
  Quality too low; estimate may be incorrect.
Average quality: -155.00% (-154%:Poor)
```
:information_source: 255 - this is the limit.

**Comment by guettli** [comment]
@zvezdochiot please elaborate. What do you want to tell me with your comment?

**Comment by zvezdochiot** [comment]
Hi @guettli .

:information_source: The value `255` is the maximum in the quantization tables (see above). Quantization tables are a jpeg filtering mechanism. It is impossible to get more filtering with jpeg, and the file will not become smaller than this level.

**Comment by guettli** [comment]
My goal is the reduce the file size while keeping the image nice for the human eye.

I found a solution and published it here: https://github.com/guettli/cheat-sheet/blob/master/README.md#reducing-image-size

`convert image.jpg -resize '1000x>' -define jpeg:extent=100KB result.jpg`

This reduced the resolution before reducing the jpg-size. The result is better than with `jpegoptim`.
